### PR TITLE
Fix crash on macOS with clang compiler

### LINKED ISSUE #17714: Segfault after Main Menu
In the latest experimental builds the game crashes with a SIGSEGV as soon as I enter character creation or load a game.
The issue is only present in latest builds, I currently play build 5194 which is version 0.C-16370-g75a3362 with no crashes.

I've been using the tiles version on OS X 10.11 (El Capitan), both from the Jenkins build, homebrew install and building from source myself.

Here is a screenshot of the last thing I see before the crash.

screen shot 2016-07-16 at 12 16 11

Here is the debug log from the game:
[debug.log.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/367216/debug.log.txt)

And here is the crash report from the OS:
[Crash Report.txt](https://github.com/CleverRaven/Cataclysm-DDA/files/367211/Crash.Report.txt)

### PR DESCRIPTION
**Pull request by ghost**
Fixes #17714.

### CODE REVIEW

**Comment by DLaboratory** [comment]
Oops, build bot just marked the check failed because Makefile is modified in this PR.

File: src/npc_class.h [hunk 1]
```diff
@@ -33,6 +33,7 @@ class distribution
 
         distribution operator+( const distribution &other ) const;
         distribution operator*( const distribution &other ) const;
+        distribution& operator=( const distribution &other );
```
Comment (BevapDin) [inline src/npc_class.h]: The copy assignment operator should be automatically created by the compiler. Does your compiler complain without this?

Comment (DLaboratory) [inline src/npc_class.h]: Here is the whole story: [https://github.com/CleverRaven/Cataclysm-DDA/issues/17714](https://github.com/CleverRaven/Cataclysm-DDA/issues/17714). I suspect clang compiler doesn't copy the `std::function` properly with optimization switch on, causes the crash.

**Comment by sxmichaels** [comment]
Excellent work, @DLaboratory — Mac users express their gratitude! But I suspect the devs might be overlooking this fix since the Makefile changes make the build fail unconditionally; maybe you could split the bugfix and Makefile alterations into two separate commits?

**Comment by DLaboratory** [comment]
Okay, changes on Makefile is reverted.

**Comment by DLaboratory** [comment]
```
diff --git a/src/npc_class.cpp b/src/npc_class.cpp
index 7a6e467..f369a55 100644
--- a/src/npc_class.cpp
+++ b/src/npc_class.cpp
@@ -383,3 +383,10 @@ distribution distribution::operator*( const distribution &other ) const
         return my_fun() * other_fun();
     } );
 }
+
+distribution &distribution::operator=( const distribution &other )
+{
+    generator_function = other.generator_function;
+    return *this;
+}
+
diff --git a/src/npc_class.h b/src/npc_class.h
index 9e0182f..98c7233 100644
--- a/src/npc_class.h
+++ b/src/npc_class.h
@@ -33,6 +33,7 @@ class distribution
 
         distribution operator+( const distribution &other ) const;
         distribution operator*( const distribution &other ) const;
+        distribution &operator=( const distribution &other );
 
         static distribution constant( float val );
         static distribution rng_roll( int from, int to );
```

**Comment by Zireael07** [comment]
Why are you closing issues before they can be reviewed/merged?

**Comment by cainiaowu** [comment]
Yelling at a ghost wont give you any reply, he deleted his account.

**Comment by Soyweiser** [comment]
Did he delete his account or github?

**Pull request closed without merging**